mindmeld/models/query_features.py

```diff
@@ -592,36 +613,37 @@ def extract_char_ngrams(lengths=(1,), thresholds=(0,), **kwargs):
     """
     del kwargs
     threshold_list = list(thresholds)
-    char_thresholds = threshold_list + [0] * (len(lengths) - len(threshold_list))
+    char_thresholds = threshold_list + [1] * (len(lengths) - len(threshold_list))
 
     def _extractor(query, resources):
-        query_text = query.normalized_text
-        ngram_counter = Counter()
 
+        query_text = re.sub(r"\d", "0", query.normalized_text)
+
+        ngram_counter = Counter()
         for length, threshold in zip(lengths, char_thresholds):
             for i in range(len(query_text) - length + 1):
                 char_ngram = []
                 for token in query_text[i : i + length]:
                     char_ngram.append(token)
-                if (
-                    resources[CHAR_NGRAM_FREQ_RSC].get("".join(char_ngram), 1)
-                    > threshold
-                ):
-                    ngram_counter.update(
-                        [
-                            "char_ngram|length:{}|ngram:{}".format(
-                                len(char_ngram), " ".join(char_ngram)
-                            )
-                        ]
-                    )
+
+                freq = resources[CHAR_NGRAM_FREQ_RSC].get("".join(char_ngram), 0)
```

Comment:
shouldnt `"".join(char_ngram)` be `" ".join(char_ngram)` ?

Reply:
To prevent such issues, lets make `" "` into a global var and call `SPACE_SEP.join(char_ngram)`

Reply:
We store the char ngrams without a space, but add a space while featurizing. Will make it uniform.